common/integration_testing/src/build_nacl.mk

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains the implementation of the ../include.mk interface that
+# builds the JS-to-C++ test runner using the Native Client toolchain.
+
+# Internal constant containing the list of additional C/C++ static libraries to
+# link against.
+#
+# Notes:
+# * INTEGRATION_TESTING_LIB has to come before CPP_COMMON_LIB, since the
+#   former uses symbols from the latter.
+# * DEFAULT_NACL_LIBS is specified twice, because there's a circular dependency
```

Comment:
Improved the comment to explain this hack, per offline discussion. Thanks for raising the question.

Reply:
After thinking more on this, I guess there's a somewhat more straightforward alternative. If I stop putting `integration_test_pp_module.cc` into a static library, and instead link directly against its object file, then its symbols will be included into the result without any tricks.

Also it looks like I need to do the same for the upcoming Emscripten anyway, for an unrelated reason though: the way how Emscripten exposes its guts to JS is via statically initialized global variables, and again the linker will just prune them away if I put it into a static library.

Reply:
Done